Source/Addendum/AdViewUtils.swift

```diff
@@ -15,10 +15,162 @@
 
 import Foundation
 import WebKit
+import StoreKit
+
+public class PrebidSKAdNetworkHelper: NSObject {
+    
+    private weak var viewController: UIViewController?
+    private weak var wkWebView: WKWebView?
+    
+    public override init() {
+        super.init()
+    }
+
+    @available(iOS 14.0, *)
+    @objc
+    public func subscribeOnAdClicked(viewController: UIViewController, adView: UIView) {
+        
+        self.viewController = viewController
+        
+        let view = AdViewUtils.findView(adView) { (subView) -> Bool in
+            return AdViewUtils.isWKWebView(subView)
+        }
+        
+        guard let wkWebView = view as? WKWebView else {
+            Log.warn("View doesn't contain WKWebView")
+            return
+            
+        }
+
+        wkWebView.navigationDelegate = self
```

Comment:
In this point, GMA SDK losses the "connection" with web view and GMA SDK is not able to process navigation events from the own web view anymore. Is it expected behavior?

Reply:
It is true 

Do you think that GAM uses `navigationDelegate` with some vital logic? If I were GAM developer I would not expect a big profit of the approach which can  be overrode by any developer out of SDK

Reply:
The MRAID and click behaviour is based on this delegate.

When user taps on the ad the rendering SDK analyses the event and performs the respective action, f.e - opens a clickthrough with the click URL or performs MRAID action.

Here is an example of this method in the rendering module
https://github.com/openx/prebid-mobile-ios/blob/feature-rendering-module/PrebidMobile/Rendering/PrebidMobileRendering/PrebidMobileRendering/AdTypes/AdView/PBMWebView.m#L277

Here is MoPub's one:
https://github.com/mopub/mopub-ios-sdk/blob/3e1382ed6686c59cf83d259f2bd8796de6c88c55/MoPubSDK/Internal/HTML/MPWebView.m#L387

Reply:
I see, then I should rollback my changes

Reply:
Have you considered the method swizzling?

https://nshipster.com/method-swizzling/

I do not really like this approach but It could be a good option in this case.

Reply:
which method do you propose to swizzle ?

Reply:
However, it  is not clear what to "swizzle" here  :(

Reply:
we can search the needed class, as described here:
https://stackoverflow.com/questions/30044366/method-swizzling-for-protocols

but it looks too tricky.